### PR TITLE
feat: activeUntil and new withdrawable impl

### PR DESCRIPTION
**Pull request by xmxanuel**
- new function `activeUntil`
- new behaviour of `withdrawable`
- active(tokenId) based on `activeUntil`
- updated deploy script
- removed temporary nftTypes array  from constructor

### CODE REVIEW

File: src/nft.sol [hunk 1]
```diff
@@ -29,6 +29,8 @@ contract FundingNFT is ERC721, Ownable {
 
     mapping (uint128 => NFTType) public nftTypes;
 
+    mapping (uint => uint64) public minted;
```
Comment (manuelpolzhofer) [inline src/nft.sol:32]: I moved it here but we could also store it in the pool contract

File: src/nft.sol [hunk 2]
```diff
@@ -125,54 +134,46 @@ contract FundingNFT is ERC721, Ownable {
     function withdrawable(uint tokenId) public view returns(uint128) {
         uint128 amtPerSec = pool.getAmtPerSecSubSender(address(this), tokenId);
         if (amtPerSec == 0) {
-            return 0;
+            return type(uint128).max;
```
Comment (CodeSandwich) [inline src/nft.sol:137]: That's generous :sweat_smile: 
```suggestion
            return pool.withdrawableSubSender(address(this), tokenId);
```

**Review by CodeSandwich — CHANGES_REQUESTED** [review]
I love how expiration option C just wiped some of the most complex logic 😌

File: src/nft.sol (hunk 2 above)
Comment (CodeSandwich) [reply src/nft.sol:137]: BTW my suggestion isn't correct either, it doesn't take into account locking funds.

Comment (manuelpolzhofer) [reply src/nft.sol]: Because you only need to lock your `amtPerSec * cycleSecs` which is always zero in that case.

**Pull request merged**